dotnet/src/IntegrationTests/Connectors/Memory/Pinecone/PineconeVectorRecordStoreTests.cs

```diff
@@ -0,0 +1,399 @@
+            }
+
+            return new PineconeHotel
+            {
+                HotelId = storageModel.Id,
+                HotelName = "Custom " + (string)storageModel.Metadata[nameof(PineconeHotel.HotelName)].Inner!,
+                Description = (string)storageModel.Metadata[nameof(PineconeHotel.Description)].Inner!,
+                HotelCode = (int)(double)storageModel.Metadata[nameof(PineconeHotel.HotelCode)].Inner!,
+                HotelRating = (float)(double)storageModel.Metadata[nameof(PineconeHotel.HotelRating)].Inner!,
+                ParkingIncluded = (bool)storageModel.Metadata["parking_is_included"].Inner!,
+                Tags = ((MetadataValue[])storageModel.Metadata[nameof(PineconeHotel.Tags)].Inner!)!.Select(x => (string)x.Inner!).ToList(),
+            };
+        }
+    }
+
+    #region Negative
+
+    [PineconeFact]
+    public void UseRecordWithNoEmbeddingThrows()
+    {
+        var exception = Assert.Throws<ArgumentException>(
+            () => new PineconeVectorRecordStore<PineconeRecordNoEmbedding>(
+            this.Fixture.Client,
+            new PineconeVectorRecordStoreOptions<PineconeRecordNoEmbedding>
+            {
+                DefaultCollectionName = "Whatever",
+                MapperType = PineconeRecordMapperType.Default
+            }));
+
+        Assert.Equal(
+            $"No vector property found on type {typeof(PineconeRecordNoEmbedding).FullName}.",
+            exception.Message);
+    }
+
+#pragma warning disable CA1812
+    private sealed record PineconeRecordNoEmbedding
+    {
+        [VectorStoreRecordKey]
+        public int Id { get; set; }
+
+        [VectorStoreRecordData]
+        public string? Name { get; set; }
+    }
+#pragma warning restore CA1812
+
+    [PineconeFact]
+    public void UseRecordWithMultipleEmbeddingsThrows()
+    {
+        var exception = Assert.Throws<ArgumentException>(
+            () => new PineconeVectorRecordStore<PineconeRecordMultipleEmbeddings>(
+            this.Fixture.Client,
+            new PineconeVectorRecordStoreOptions<PineconeRecordMultipleEmbeddings>
+            {
+                DefaultCollectionName = "Whatever",
+                MapperType = PineconeRecordMapperType.Default
+            }));
+
+        Assert.Equal(
+            $"Multiple vector properties found on type {typeof(PineconeRecordMultipleEmbeddings).FullName} while only one is supported.",
+            exception.Message);
+    }
+
+#pragma warning disable CA1812
+    private sealed record PineconeRecordMultipleEmbeddings
+    {
+        [VectorStoreRecordKey]
+        public string Id { get; set; } = null!;
+
+        [VectorStoreRecordVector]
+        public ReadOnlyMemory<float> Embedding1 { get; set; }
+
+        [VectorStoreRecordVector]
+        public ReadOnlyMemory<float> Embedding2 { get; set; }
+    }
+#pragma warning restore CA1812
+
+    [PineconeFact]
+    public void UseRecordWithUnsupportedKeyTypeThrows()
```

Comment:
Having these is of course not bad, but there may be overlap with the unit tests in `VectorStoreRecordPropertyReaderTests` in case you prefer to reduce the number of cases.